proto/topology.proto

```diff
@@ -98,7 +98,7 @@ message ZoneData {
 
 message ServerData {
     required uint32 serverId = 1;
-    required string hostName = 2;
+    optional string hostName = 2;
```

Comment:
This requirement is strange. When hostName is not available, you can use ip instead. HostName is just an identifier internally. There is no need to change it to optional. Discuss with the issue author whether we should change it this way.

Reply:
While deploying the bs,  not setting the hostname will fail. So change it to optional?